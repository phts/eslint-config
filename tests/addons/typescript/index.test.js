'use strict'

const helpers = require('../../helpers')

describe('addons/typescript', () => {
  helpers.itGeneratesCorrectConfig({
    extends: ['index', 'addons/typescript'],
    files: ['target.ts', 'target.tsx', 'dir/target.ts', 'dir/target.tsx'],
    for: '*.{ts,tsx}',
  })

  helpers.itGeneratesCorrectConfig({
    extends: ['index', 'addons/typescript'],
    files: ['target.d.ts', 'dir/target.d.ts'],
    for: '*.d.ts',
  })
})
