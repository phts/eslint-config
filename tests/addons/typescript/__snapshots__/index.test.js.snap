// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addons/typescript generates correct config for *.{ts,tsx} files 1`] = `
{
  "env": {
    "es6": true,
  },
  "globals": {},
  "ignorePatterns": [
    "playground/",
  ],
  "noInlineConfig": undefined,
  "parser": "node_modules/@typescript-eslint/parser/dist/index.js",
  "parserOptions": {
    "project": "./tsconfig.json",
    "sourceType": "module",
    "tsconfigRootDir": ".",
  },
  "plugins": [
    "eslint-comments",
    "import",
    "@typescript-eslint",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": {
    "@typescript-eslint/adjacent-overload-signatures": [
      "error",
    ],
    "@typescript-eslint/array-type": [
      "error",
      {
        "default": "array-simple",
      },
    ],
    "@typescript-eslint/ban-ts-comment": [
      "error",
    ],
    "@typescript-eslint/ban-ts-ignore": [
      "off",
    ],
    "@typescript-eslint/ban-types": [
      "error",
    ],
    "@typescript-eslint/consistent-type-assertions": [
      "error",
      {
        "assertionStyle": "as",
        "objectLiteralTypeAssertions": "allow-as-parameter",
      },
    ],
    "@typescript-eslint/consistent-type-definitions": [
      "error",
      "interface",
    ],
    "@typescript-eslint/explicit-function-return-type": [
      "off",
    ],
    "@typescript-eslint/explicit-member-accessibility": [
      "off",
    ],
    "@typescript-eslint/indent": [
      "error",
      2,
    ],
    "@typescript-eslint/member-delimiter-style": [
      "error",
      {
        "multiline": {
          "delimiter": "none",
        },
        "singleline": {
          "delimiter": "comma",
          "requireLast": false,
        },
      },
    ],
    "@typescript-eslint/member-ordering": [
      "error",
    ],
    "@typescript-eslint/no-array-constructor": [
      "error",
    ],
    "@typescript-eslint/no-empty-function": [
      "error",
    ],
    "@typescript-eslint/no-empty-interface": [
      "off",
    ],
    "@typescript-eslint/no-explicit-any": [
      "error",
    ],
    "@typescript-eslint/no-extra-non-null-assertion": [
      "error",
    ],
    "@typescript-eslint/no-extra-semi": [
      "error",
    ],
    "@typescript-eslint/no-extraneous-class": [
      "error",
    ],
    "@typescript-eslint/no-inferrable-types": [
      "error",
    ],
    "@typescript-eslint/no-loss-of-precision": [
      "error",
    ],
    "@typescript-eslint/no-misused-new": [
      "error",
    ],
    "@typescript-eslint/no-namespace": [
      "error",
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": [
      "error",
    ],
    "@typescript-eslint/no-non-null-assertion": [
      "off",
    ],
    "@typescript-eslint/no-parameter-properties": [
      "off",
    ],
    "@typescript-eslint/no-this-alias": [
      "error",
      {
        "allowDestructuring": true,
      },
    ],
    "@typescript-eslint/no-unnecessary-type-constraint": [
      "error",
    ],
    "@typescript-eslint/no-unused-vars": [
      "error",
    ],
    "@typescript-eslint/no-var-requires": [
      "error",
    ],
    "@typescript-eslint/prefer-as-const": [
      "error",
    ],
    "@typescript-eslint/prefer-namespace-keyword": [
      "error",
    ],
    "@typescript-eslint/restrict-plus-operands": [
      "error",
    ],
    "@typescript-eslint/triple-slash-reference": [
      "error",
    ],
    "accessor-pairs": [
      "error",
    ],
    "array-bracket-spacing": [
      "error",
    ],
    "brace-style": [
      "error",
    ],
    "camelcase": [
      "error",
    ],
    "comma-dangle": [
      "error",
      "always-multiline",
    ],
    "comma-spacing": [
      "error",
    ],
    "comma-style": [
      "error",
    ],
    "constructor-super": [
      "off",
    ],
    "curly": [
      "error",
    ],
    "dot-location": [
      "error",
      "property",
    ],
    "eol-last": [
      "error",
    ],
    "eqeqeq": [
      "error",
    ],
    "eslint-comments/disable-enable-pair": [
      "error",
      {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": [
      "error",
    ],
    "eslint-comments/no-unused-disable": [
      "error",
    ],
    "eslint-comments/no-unused-enable": [
      "error",
    ],
    "for-direction": [
      "error",
    ],
    "func-call-spacing": [
      "error",
    ],
    "getter-return": [
      "off",
    ],
    "import/export": [
      "error",
    ],
    "import/first": [
      "error",
    ],
    "import/named": [
      "off",
    ],
    "import/newline-after-import": [
      "error",
    ],
    "import/no-absolute-path": [
      "error",
    ],
    "import/no-cycle": [
      "error",
    ],
    "import/no-named-as-default": [
      "error",
    ],
    "import/no-named-as-default-member": [
      "error",
    ],
    "import/no-self-import": [
      "error",
    ],
    "import/no-useless-path-segments": [
      "error",
    ],
    "import/order": [
      "error",
    ],
    "indent": [
      "off",
      2,
    ],
    "jsx-quotes": [
      "error",
      "prefer-double",
    ],
    "key-spacing": [
      "error",
      {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": [
      "error",
    ],
    "max-len": [
      "error",
      {
        "code": 100,
      },
    ],
    "new-cap": [
      "error",
    ],
    "new-parens": [
      "error",
    ],
    "no-array-constructor": [
      "off",
    ],
    "no-async-promise-executor": [
      "error",
    ],
    "no-caller": [
      "error",
    ],
    "no-case-declarations": [
      "error",
    ],
    "no-class-assign": [
      "error",
    ],
    "no-compare-neg-zero": [
      "error",
    ],
    "no-cond-assign": [
      "error",
    ],
    "no-console": [
      "error",
      {
        "allow": [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": [
      "off",
    ],
    "no-constant-condition": [
      "error",
    ],
    "no-control-regex": [
      "error",
    ],
    "no-debugger": [
      "error",
    ],
    "no-delete-var": [
      "error",
    ],
    "no-dupe-args": [
      "off",
    ],
    "no-dupe-class-members": [
      "off",
    ],
    "no-dupe-else-if": [
      "error",
    ],
    "no-dupe-keys": [
      "off",
    ],
    "no-duplicate-case": [
      "error",
    ],
    "no-duplicate-imports": [
      "error",
    ],
    "no-empty": [
      "error",
    ],
    "no-empty-character-class": [
      "error",
    ],
    "no-empty-function": [
      "off",
    ],
    "no-empty-pattern": [
      "error",
    ],
    "no-eval": [
      "error",
    ],
    "no-ex-assign": [
      "error",
    ],
    "no-extra-bind": [
      "error",
    ],
    "no-extra-boolean-cast": [
      "error",
    ],
    "no-extra-parens": [
      "off",
      "all",
      {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": [
      "off",
    ],
    "no-fallthrough": [
      "error",
    ],
    "no-floating-decimal": [
      "error",
    ],
    "no-func-assign": [
      "off",
    ],
    "no-global-assign": [
      "error",
    ],
    "no-implied-eval": [
      "error",
    ],
    "no-import-assign": [
      "off",
    ],
    "no-inner-declarations": [
      "error",
    ],
    "no-invalid-regexp": [
      "error",
    ],
    "no-irregular-whitespace": [
      "error",
    ],
    "no-label-var": [
      "error",
    ],
    "no-labels": [
      "error",
    ],
    "no-lone-blocks": [
      "error",
    ],
    "no-loss-of-precision": [
      "off",
    ],
    "no-misleading-character-class": [
      "error",
    ],
    "no-mixed-spaces-and-tabs": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "error",
    ],
    "no-new-object": [
      "error",
    ],
    "no-new-require": [
      "error",
    ],
    "no-new-symbol": [
      "off",
    ],
    "no-nonoctal-decimal-escape": [
      "error",
    ],
    "no-obj-calls": [
      "off",
    ],
    "no-octal": [
      "error",
    ],
    "no-path-concat": [
      "error",
    ],
    "no-proto": [
      "error",
    ],
    "no-prototype-builtins": [
      "error",
    ],
    "no-redeclare": [
      "off",
    ],
    "no-regex-spaces": [
      "error",
    ],
    "no-return-assign": [
      "error",
    ],
    "no-return-await": [
      "error",
    ],
    "no-self-assign": [
      "error",
    ],
    "no-self-compare": [
      "error",
    ],
    "no-sequences": [
      "error",
    ],
    "no-setter-return": [
      "off",
    ],
    "no-shadow-restricted-names": [
      "error",
    ],
    "no-sparse-arrays": [
      "error",
    ],
    "no-tabs": [
      "error",
    ],
    "no-this-before-super": [
      "off",
    ],
    "no-throw-literal": [
      "error",
    ],
    "no-trailing-spaces": [
      "error",
    ],
    "no-undef": [
      "off",
    ],
    "no-undef-init": [
      "error",
    ],
    "no-unexpected-multiline": [
      "error",
    ],
    "no-unmodified-loop-condition": [
      "error",
    ],
    "no-unneeded-ternary": [
      "error",
    ],
    "no-unreachable": [
      "off",
    ],
    "no-unsafe-finally": [
      "error",
    ],
    "no-unsafe-negation": [
      "off",
    ],
    "no-unsafe-optional-chaining": [
      "error",
    ],
    "no-unused-labels": [
      "error",
    ],
    "no-unused-vars": [
      "off",
    ],
    "no-use-before-define": [
      "error",
    ],
    "no-useless-backreference": [
      "error",
    ],
    "no-useless-call": [
      "error",
    ],
    "no-useless-catch": [
      "error",
    ],
    "no-useless-computed-key": [
      "error",
    ],
    "no-useless-constructor": [
      "error",
    ],
    "no-useless-escape": [
      "error",
    ],
    "no-useless-rename": [
      "error",
    ],
    "no-var": [
      "error",
    ],
    "no-whitespace-before-property": [
      "error",
    ],
    "no-with": [
      "error",
    ],
    "object-curly-spacing": [
      "error",
      "never",
    ],
    "object-property-newline": [
      "error",
      {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": [
      "error",
    ],
    "padded-blocks": [
      "error",
      "never",
    ],
    "prefer-const": [
      "error",
    ],
    "prefer-rest-params": [
      "error",
    ],
    "prefer-spread": [
      "error",
    ],
    "quote-props": [
      "error",
      "as-needed",
    ],
    "quotes": [
      "error",
      "single",
      {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "require-yield": [
      "error",
    ],
    "rest-spread-spacing": [
      "error",
    ],
    "semi": [
      "error",
      "never",
    ],
    "semi-spacing": [
      "error",
    ],
    "space-before-blocks": [
      "error",
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-in-parens": [
      "error",
    ],
    "space-infix-ops": [
      "error",
    ],
    "space-unary-ops": [
      "error",
    ],
    "spaced-comment": [
      "error",
    ],
    "strict": [
      "error",
    ],
    "template-curly-spacing": [
      "error",
    ],
    "use-isnan": [
      "error",
    ],
    "valid-typeof": [
      "off",
    ],
    "wrap-iife": [
      "error",
    ],
    "yield-star-spacing": [
      "error",
      "before",
    ],
    "yoda": [
      "error",
    ],
  },
  "settings": {},
}
`;

exports[`addons/typescript generates correct config for *.d.ts files 1`] = `
{
  "env": {
    "es6": true,
  },
  "globals": {},
  "ignorePatterns": [
    "playground/",
  ],
  "noInlineConfig": undefined,
  "parser": "node_modules/@typescript-eslint/parser/dist/index.js",
  "parserOptions": {
    "project": "./tsconfig.json",
    "sourceType": "module",
    "tsconfigRootDir": ".",
  },
  "plugins": [
    "eslint-comments",
    "import",
    "@typescript-eslint",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": {
    "@typescript-eslint/adjacent-overload-signatures": [
      "error",
    ],
    "@typescript-eslint/array-type": [
      "error",
      {
        "default": "array-simple",
      },
    ],
    "@typescript-eslint/ban-ts-comment": [
      "error",
    ],
    "@typescript-eslint/ban-ts-ignore": [
      "off",
    ],
    "@typescript-eslint/ban-types": [
      "error",
    ],
    "@typescript-eslint/consistent-type-assertions": [
      "error",
      {
        "assertionStyle": "as",
        "objectLiteralTypeAssertions": "allow-as-parameter",
      },
    ],
    "@typescript-eslint/consistent-type-definitions": [
      "error",
      "interface",
    ],
    "@typescript-eslint/explicit-function-return-type": [
      "off",
    ],
    "@typescript-eslint/explicit-member-accessibility": [
      "off",
    ],
    "@typescript-eslint/indent": [
      "error",
      2,
    ],
    "@typescript-eslint/member-delimiter-style": [
      "error",
      {
        "multiline": {
          "delimiter": "none",
        },
        "singleline": {
          "delimiter": "comma",
          "requireLast": false,
        },
      },
    ],
    "@typescript-eslint/member-ordering": [
      "error",
    ],
    "@typescript-eslint/no-array-constructor": [
      "error",
    ],
    "@typescript-eslint/no-empty-function": [
      "error",
    ],
    "@typescript-eslint/no-empty-interface": [
      "off",
    ],
    "@typescript-eslint/no-explicit-any": [
      "off",
    ],
    "@typescript-eslint/no-extra-non-null-assertion": [
      "error",
    ],
    "@typescript-eslint/no-extra-semi": [
      "error",
    ],
    "@typescript-eslint/no-extraneous-class": [
      "error",
    ],
    "@typescript-eslint/no-inferrable-types": [
      "error",
    ],
    "@typescript-eslint/no-loss-of-precision": [
      "error",
    ],
    "@typescript-eslint/no-misused-new": [
      "error",
    ],
    "@typescript-eslint/no-namespace": [
      "error",
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": [
      "error",
    ],
    "@typescript-eslint/no-non-null-assertion": [
      "off",
    ],
    "@typescript-eslint/no-parameter-properties": [
      "off",
    ],
    "@typescript-eslint/no-this-alias": [
      "error",
      {
        "allowDestructuring": true,
      },
    ],
    "@typescript-eslint/no-unnecessary-type-constraint": [
      "error",
    ],
    "@typescript-eslint/no-unused-vars": [
      "error",
    ],
    "@typescript-eslint/no-var-requires": [
      "error",
    ],
    "@typescript-eslint/prefer-as-const": [
      "error",
    ],
    "@typescript-eslint/prefer-namespace-keyword": [
      "error",
    ],
    "@typescript-eslint/restrict-plus-operands": [
      "error",
    ],
    "@typescript-eslint/triple-slash-reference": [
      "error",
    ],
    "accessor-pairs": [
      "error",
    ],
    "array-bracket-spacing": [
      "error",
    ],
    "brace-style": [
      "error",
    ],
    "camelcase": [
      "error",
    ],
    "comma-dangle": [
      "error",
      "always-multiline",
    ],
    "comma-spacing": [
      "error",
    ],
    "comma-style": [
      "error",
    ],
    "constructor-super": [
      "off",
    ],
    "curly": [
      "error",
    ],
    "dot-location": [
      "error",
      "property",
    ],
    "eol-last": [
      "error",
    ],
    "eqeqeq": [
      "error",
    ],
    "eslint-comments/disable-enable-pair": [
      "error",
      {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": [
      "error",
    ],
    "eslint-comments/no-unused-disable": [
      "error",
    ],
    "eslint-comments/no-unused-enable": [
      "error",
    ],
    "for-direction": [
      "error",
    ],
    "func-call-spacing": [
      "error",
    ],
    "getter-return": [
      "off",
    ],
    "import/export": [
      "error",
    ],
    "import/first": [
      "error",
    ],
    "import/named": [
      "off",
    ],
    "import/newline-after-import": [
      "error",
    ],
    "import/no-absolute-path": [
      "error",
    ],
    "import/no-cycle": [
      "error",
    ],
    "import/no-named-as-default": [
      "error",
    ],
    "import/no-named-as-default-member": [
      "error",
    ],
    "import/no-self-import": [
      "error",
    ],
    "import/no-useless-path-segments": [
      "error",
    ],
    "import/order": [
      "error",
    ],
    "indent": [
      "off",
      2,
    ],
    "jsx-quotes": [
      "error",
      "prefer-double",
    ],
    "key-spacing": [
      "error",
      {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": [
      "error",
    ],
    "max-len": [
      "error",
      {
        "code": 100,
      },
    ],
    "new-cap": [
      "error",
    ],
    "new-parens": [
      "error",
    ],
    "no-array-constructor": [
      "off",
    ],
    "no-async-promise-executor": [
      "error",
    ],
    "no-caller": [
      "error",
    ],
    "no-case-declarations": [
      "error",
    ],
    "no-class-assign": [
      "error",
    ],
    "no-compare-neg-zero": [
      "error",
    ],
    "no-cond-assign": [
      "error",
    ],
    "no-console": [
      "error",
      {
        "allow": [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": [
      "off",
    ],
    "no-constant-condition": [
      "error",
    ],
    "no-control-regex": [
      "error",
    ],
    "no-debugger": [
      "error",
    ],
    "no-delete-var": [
      "error",
    ],
    "no-dupe-args": [
      "off",
    ],
    "no-dupe-class-members": [
      "off",
    ],
    "no-dupe-else-if": [
      "error",
    ],
    "no-dupe-keys": [
      "off",
    ],
    "no-duplicate-case": [
      "error",
    ],
    "no-duplicate-imports": [
      "error",
    ],
    "no-empty": [
      "error",
    ],
    "no-empty-character-class": [
      "error",
    ],
    "no-empty-function": [
      "off",
    ],
    "no-empty-pattern": [
      "error",
    ],
    "no-eval": [
      "error",
    ],
    "no-ex-assign": [
      "error",
    ],
    "no-extra-bind": [
      "error",
    ],
    "no-extra-boolean-cast": [
      "error",
    ],
    "no-extra-parens": [
      "off",
      "all",
      {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": [
      "off",
    ],
    "no-fallthrough": [
      "error",
    ],
    "no-floating-decimal": [
      "error",
    ],
    "no-func-assign": [
      "off",
    ],
    "no-global-assign": [
      "error",
    ],
    "no-implied-eval": [
      "error",
    ],
    "no-import-assign": [
      "off",
    ],
    "no-inner-declarations": [
      "error",
    ],
    "no-invalid-regexp": [
      "error",
    ],
    "no-irregular-whitespace": [
      "error",
    ],
    "no-label-var": [
      "error",
    ],
    "no-labels": [
      "error",
    ],
    "no-lone-blocks": [
      "error",
    ],
    "no-loss-of-precision": [
      "off",
    ],
    "no-misleading-character-class": [
      "error",
    ],
    "no-mixed-spaces-and-tabs": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "error",
    ],
    "no-new-object": [
      "error",
    ],
    "no-new-require": [
      "error",
    ],
    "no-new-symbol": [
      "off",
    ],
    "no-nonoctal-decimal-escape": [
      "error",
    ],
    "no-obj-calls": [
      "off",
    ],
    "no-octal": [
      "error",
    ],
    "no-path-concat": [
      "error",
    ],
    "no-proto": [
      "error",
    ],
    "no-prototype-builtins": [
      "error",
    ],
    "no-redeclare": [
      "off",
    ],
    "no-regex-spaces": [
      "error",
    ],
    "no-return-assign": [
      "error",
    ],
    "no-return-await": [
      "error",
    ],
    "no-self-assign": [
      "error",
    ],
    "no-self-compare": [
      "error",
    ],
    "no-sequences": [
      "error",
    ],
    "no-setter-return": [
      "off",
    ],
    "no-shadow-restricted-names": [
      "error",
    ],
    "no-sparse-arrays": [
      "error",
    ],
    "no-tabs": [
      "error",
    ],
    "no-this-before-super": [
      "off",
    ],
    "no-throw-literal": [
      "error",
    ],
    "no-trailing-spaces": [
      "error",
    ],
    "no-undef": [
      "off",
    ],
    "no-undef-init": [
      "error",
    ],
    "no-unexpected-multiline": [
      "error",
    ],
    "no-unmodified-loop-condition": [
      "error",
    ],
    "no-unneeded-ternary": [
      "error",
    ],
    "no-unreachable": [
      "off",
    ],
    "no-unsafe-finally": [
      "error",
    ],
    "no-unsafe-negation": [
      "off",
    ],
    "no-unsafe-optional-chaining": [
      "error",
    ],
    "no-unused-labels": [
      "error",
    ],
    "no-unused-vars": [
      "off",
    ],
    "no-use-before-define": [
      "error",
    ],
    "no-useless-backreference": [
      "error",
    ],
    "no-useless-call": [
      "error",
    ],
    "no-useless-catch": [
      "error",
    ],
    "no-useless-computed-key": [
      "error",
    ],
    "no-useless-constructor": [
      "error",
    ],
    "no-useless-escape": [
      "error",
    ],
    "no-useless-rename": [
      "error",
    ],
    "no-var": [
      "error",
    ],
    "no-whitespace-before-property": [
      "error",
    ],
    "no-with": [
      "error",
    ],
    "object-curly-spacing": [
      "error",
      "never",
    ],
    "object-property-newline": [
      "error",
      {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": [
      "error",
    ],
    "padded-blocks": [
      "error",
      "never",
    ],
    "prefer-const": [
      "error",
    ],
    "prefer-rest-params": [
      "error",
    ],
    "prefer-spread": [
      "error",
    ],
    "quote-props": [
      "error",
      "as-needed",
    ],
    "quotes": [
      "error",
      "single",
      {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "require-yield": [
      "error",
    ],
    "rest-spread-spacing": [
      "error",
    ],
    "semi": [
      "error",
      "never",
    ],
    "semi-spacing": [
      "error",
    ],
    "space-before-blocks": [
      "error",
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-in-parens": [
      "error",
    ],
    "space-infix-ops": [
      "error",
    ],
    "space-unary-ops": [
      "error",
    ],
    "spaced-comment": [
      "error",
    ],
    "strict": [
      "error",
    ],
    "template-curly-spacing": [
      "error",
    ],
    "use-isnan": [
      "error",
    ],
    "valid-typeof": [
      "off",
    ],
    "wrap-iife": [
      "error",
    ],
    "yield-star-spacing": [
      "error",
      "before",
    ],
    "yoda": [
      "error",
    ],
  },
  "settings": {},
}
`;
